common/utilities/secrets.py

```diff
@@ -0,0 +1,51 @@
+"""
+Secrets
+
+This module holds the secret config used by an application. To use this module, first
+call `setup_secrets_config` to populate `secret_config`. Then, just get any required config
+using `get_secret_config` (or directly use `secret_config` for more complex use cases).
+"""
+import logging
+import os
+from typing import Dict, Optional
+
+secret_config: Dict[str, str] = {}
+
```

Comment:
So something that Joe raised while he was having a run through our project was that this config stuff seemed a bit 'magic', and (at least in other languages) probably should be something that should be wrapped in a class with static elements (though obviously python doesn't have a natural static attributes in classes). In particular if you were to import the methods and not the module the element retrieval fails. I don't feel too strongly on changing the implementation but I figured it's something to be wary of.

Reply:
Ok, I've thought about this a bit.

When implementing this, the approach was to go with the simplest thing that gets the job done. So just filter environment variables by a prefix, have a setup function that populates a a dict, then that dict has all the config in it that everything else can reference (I thought about just using `os.environ` but that would add a little duplication everywhere). Then the function to get a config value was added, as a helper really (that does a bit of logging).

Could this be done as a class? Yes. But I don't think it would be any simpler (you'd probably want a singleton, but then would this be dependency-injected everywhere or just imported (ie dependency-injection in another way), in which case there's a little more complexity.

And tbh, I think modules are a bit like a singleton class. In Python, modules get cached after being imported. You can reload a module, but that's advanced wizardry that no one uses unless they really need to.

If this was a situation where multiple parts of the codebase were modifying the `config`/`secret_config` variables, then I'd find this iffy (though looking at the Python standard library, that's the way some of it's API actually works). But the only place that is doing so is this module itself. And the only place that calls the `setup_config` functions are the `main.py` files.

So I'd rather keep it as-is than change it, but if we were doing more fancy stuff, then it might need to be changed.